test/health_checker_unittest.cc

```diff
@@ -0,0 +1,89 @@
+#include "../src/health_checker.h"
+#include "gtest/gtest.h"
+#include <stdio.h>
+#include <sstream>
+#include <boost/filesystem.hpp>
+
+
+namespace google {
+
+class HealthCheckerUnittest : public ::testing::Test {
+ public:
+  HealthCheckerUnittest() { }
+
+  ~HealthCheckerUnittest() {
+    if (isolation_path_.length() > 0) {
+      std::ostringstream oss;
+      oss << isolation_path_ << "/metadata_agent_unhealthy";
+      std::remove(oss.str().c_str());
+
+      oss.str("");
+      oss.clear();
+      oss << isolation_path_ << "/kubernetes_node_thread";
+      std::remove(oss.str().c_str());
+
+      oss.str("");
+      oss.clear();
+      oss << isolation_path_ << "/kubernetes_pod_thread";
+      std::remove(oss.str().c_str());
+
+      std::remove(isolation_path_.c_str());
+    }
+  }
+ protected:
+  void SetUnhealthy(HealthChecker* healthChecker, const std::string& state_name) {
```

Comment:
Just realized that these helpers can be `static`.

Reply:
Changed